shell.nix

```diff
@@ -1,50 +1,66 @@
-{ pkgs ? import <nixpkgs> { } }:
-with pkgs;
-let
-  go = go_1_19;
-  postgresql = postgresql_14;
-  nodejs = nodejs-16_x;
-  nodePackages = pkgs.nodePackages.override { inherit nodejs; };
+{
+  pkgs ? import <nixpkgs> {},
+  dapp ? let
+    lock = (builtins.fromJSON (builtins.readFile ./flake.lock)).nodes.dapp.locked;
+  in
+    import (fetchTarball {
+      url = "https://github.com/dapphub/dapptools/archive/${lock.rev}.tar.gz";
+      sha256 = lock.narHash;
+    }) {inherit (pkgs) system;},
+  foundry-bin ? let
+    lock = (builtins.fromJSON (builtins.readFile ./flake.lock)).nodes.foundry.locked;
+  in
+    import (fetchTarball {
+        url = "https://github.com/shazow/foundry.nix/archive/${lock.rev}.tar.gz";
+        sha256 = lock.narHash;
+      }
+      + "/foundry-bin") {inherit pkgs;},
+}: let
+  go = pkgs.go_1_20;
+  postgresql = pkgs.postgresql_14;
```

Comment:
Should this be 11 since we still support postgres 11? Using a newer version might lead to using functions that aren't available.

Reply:
This is for the development environment; I think it's wise to use a more or less recent version for development, and minimum compatibility be checked in CI or somewhere else.

Reply:
If we had that validation in place that would be one thing but we do not. It will hopefully be added this quarter at which point I would feel a lot better about using 14 for development.

Reply:
:heavy_check_mark:

Reply:
Unfortunately, I had to revert the change for the dev env: logpoller's tests contain some DEFERRED clauses which fail on postgres 11 on missing fkey, but suceed on postgres 14 (didn't test intermediary versions).